projects/ng-event-plugins/src/plugins/debounce.plugin.ts

```diff
@@ -0,0 +1,64 @@
+import type {EventManager} from '@angular/platform-browser';
+
+import type {EventManagerPlugin} from '../types/event-manager-plugin';
+
+export class DebounceEventPlugin implements EventManagerPlugin {
+    private readonly regExp = /\.debounce-(?<time>\d+)(?<units>ms|s)/;
+
+    public manager!: EventManager;
+
+    public addEventListener(
+        element: HTMLElement,
+        eventName: string,
+        handler: Function,
+    ): Function {
+        let timeout: ReturnType<typeof setTimeout> | undefined;
+
+        return this.manager.addEventListener(
+            element,
+            this.unwrap(eventName),
+            (event: Event): void => {
+                clearTimeout(timeout);
+
+                timeout = setTimeout(() => {
+                    handler(event);
+
+                    timeout = undefined;
+                }, this.getTimeout(eventName));
+            },
+        );
+    }
+
+    public supports(event: string): boolean {
+        return this.regExp.test(event);
+    }
+
+    private getTimeout(event: string): number {
+        const match = this.regExp.exec(event);
+
+        if (!match?.groups) {
+            throw new Error(`Invalid event: ${event}`);
+        }
+
+        const {time, units} = match.groups;
+
+        switch (units) {
+            case 'ms':
+                return Number(time);
+            case 's':
+                return Number(time) * 1000;
+            default:
+                throw new Error(`Invalid event: ${event}`);
+        }
+    }
+
+    private unwrap(event: string): string {
+        const newEvent = event.replace(this.regExp, '');
+
+        if (this.supports(newEvent)) {
+            throw new Error('.debounce modifier can be applied only once');
```

Comment:
I'm not sure we want to dedicate code to this, seems obvious not to use it multiple times